src/connection_string.ts

```diff
@@ -255,6 +255,15 @@ export function parseOptions(
     mongoClient = undefined;
   }
 
+  // validate BSONOptions
+  if (options.useBigInt64 && !(options.promoteLongs ?? true)) {
```

Comment:
This works, but I think we usually check `typeof` because it's more explicit
```suggestion
  if (options.useBigInt64 && typeof options.promoteLongs === 'boolean' && options.promoteLongs) {
```